src/averageAges.js

```diff
@@ -55,7 +78,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people
+    .filter(person => people.find(child => person.name === child.mother));
+
+  const children = !onlyWithSon
+    ? people.filter(person => person.mother !== null)
+    : people.filter(person => person.mother !== null && person.sex === 'm');
+
+  let childrenCount = children.length;
+
+  return children.reduce((sum, child) => {
+    const currentMother = mothers.find(mother => child.mother === mother.name);
```

Comment:
Can you use here people array instead of mothers? There will be -1 loop, you will delete the loop on 81 
```suggestion
    const currentMother = people.find(mother => child.mother === mother.name);
```